test/orm/test_query.py

```diff
@@ -5820,6 +5821,25 @@ def instances(self, result, ctx):
         q1 = sess.query(User).execution_options(**execution_options)
         q1.all()
 
+    def test_options_before_compile(self):
+        User = self.classes.User
+
+        @event.listens_for(Query, "before_compile", retval=True)
```

Comment:
**mike bayer** (zzzeek) wrote:

so this event on Query globally won't be torn down in this fixture.  if we can move this test into test/orm/test_events.py -> QueryEventsTest, that would be the appropriate place for this.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2081

Reply:
Ah I see, thanks Mike. I think this explains the numerous test_schema.db files generated locally. Nevertheless I have moved the test to the file mentioned in your review as well as renamed it for consistency/clarity.

Kind Regards,
Ramon

Reply:
test_schema.db files are not related to this test directly but if you ctrl-C full runs of the test suite those files might be left over.   they are supposed to be cleaned up.